src/geometries/TubeGeometry.js

```diff
@@ -161,16 +166,49 @@ function TubeBufferGeometry( path, tubularSegments, radius, radialSegments, clos
 			normal.z = ( cos * N.z + sin * B.z );
 			normal.normalize();
 
-			normals.push( normal.x, normal.y, normal.z );
-
 			// vertex
 
-			vertex.x = P.x + radius * normal.x;
-			vertex.y = P.y + radius * normal.y;
-			vertex.z = P.z + radius * normal.z;
+			var r = radius instanceof Path ? radius.getPointAt( i / tubularSegments ).y : radius;
+			vertex.x = P.x + r * normal.x;
+			vertex.y = P.y + r * normal.y;
+			vertex.z = P.z + r * normal.z;
 
 			vertices.push( vertex.x, vertex.y, vertex.z );
 
+			// adjust normals if the radius is a Path
+
+			if ( radius instanceof Path ) {
+
+				var delta = 0.0001;
+				var t1 = i / tubularSegments - delta;
+				var t2 = i / tubularSegments + delta;
+
+				// Capping in case of danger
+
+				if ( t1 < 0 ) t1 = 0;
+				if ( t2 > 1 ) t2 = 1;
+
+				path.getPointAt( t1, P1 );
+				path.getPointAt( t2, P2 );
+				var delta = P1.distanceTo( P2 );
+
+				var r1 = radius.getPointAt( t1 ).y;
+				var r2 = radius.getPointAt( t2 ).y;
+
+				// prevent dividing by zero
```

Comment:
How was that happening?

Reply:
It shouldn't happen, but I had cases in which delta could be zero. But that was before settling on that way to calculate the offsets, so you're right, it's redundant. Removed.